ngraph/test/util/test_environment.hpp

```diff
@@ -0,0 +1,119 @@
+//*****************************************************************************
+// Copyright 2017-2020 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+#pragma once
+
+#include <iostream>
+#include <map>
+#include "gtest/gtest.h"
+#include "ngraph/ngraph.hpp"
+
+namespace ngraph
+{
+    namespace test
+    {
+        class TestEnvironment;
+
+        class Summary;
+
+        class SummaryDestroyer
+        {
+        private:
+            Summary* p_instance;
+
+        public:
+            ~SummaryDestroyer();
+
+            void initialize(Summary* p);
+        };
+
+        struct PassRate
+        {
+            enum Statuses
+            {
+                PASSED,
+                FAILED,
+                SKIPPED
+            };
+            unsigned long passed = 0;
+            unsigned long failed = 0;
+            unsigned long skipped = 0;
+
+            PassRate() = default;
+
+            PassRate(unsigned long p, unsigned long f, unsigned long s)
+            {
+                passed = p;
+                failed = f;
+                skipped = s;
+            }
+
+            float getPassrate() const
+            {
+                if (passed == 0 && failed == 0)
+                {
+                    return 0.;
+                }
+                else if (passed != 0 && failed == 0)
+                {
+                    return 100.;
+                }
+                else
+                {
+                    return (passed / (passed + failed)) * 100.;
```

Comment:
Are you sure you will get values different than 0 and 100?
I'll suggest to write it like this:
```
return 100. * passed / (passed + failed);
```
https://godbolt.org/z/8qa3an